### PR TITLE
tcti: refactor PlatformCommand function

### PR DESCRIPTION
**Pull request by tstruk**
Refactor PlatformCommand to make it complient with coding convention.
Add tcti prefix to the function name.
Rename platformcommand.c to tcti_platform_command.c

### CODE REVIEW

**Comment by AndreasFuchsSIT** [comment]
Looks good but will need rework after #843 I assume...
Since #834 is the larger PR, I recommend reworking this PR after merging the other one...

File: Makefile-test.am [hunk 1]
```diff
@@ -126,7 +126,7 @@ test_unit_tcti_device_SOURCES = test/unit/tcti-device.c tcti/tcti_device.c log/l
 test_unit_tcti_socket_CFLAGS  = $(CMOCKA_CFLAGS) $(AM_CFLAGS) $(URIPARSER_CFLAGS)
 test_unit_tcti_socket_LDADD   = $(CMOCKA_LIBS) $(libmarshal) $(URIPARSER_LIBS) $(libutil)
 test_unit_tcti_socket_LDFLAGS = -Wl,--wrap=connect,--wrap=read,--wrap=select,--wrap=write
-test_unit_tcti_socket_SOURCES = tcti/platformcommand.c tcti/tcti_socket.c \
+test_unit_tcti_socket_SOURCES = tcti/tcti_platform_command.c tcti/tcti_socket.c \
```
Comment (flihp) [inline Makefile-test.am]: @tstruk : Any objections to putting the `tcti_platform_command` function directly in the `tcti_socket` module? It's the only function in that `c` file and the implementation requires access to the private portions of the socket TCTI context so it's very tightly coupled to the socket TCTI module.

Comment (tstruk) [reply Makefile-test.am]: @flihp no problem I will do that after #843 will go in. This will need refactoring on top of that anyway.

**Comment by tstruk** [comment]
Rebased on top of #843

**Pull request merged**